core/dbt/task/compile.py

```diff
@@ -22,3 +31,67 @@ def get_runner_type(self):
 
     def task_end_messages(self, results):
         dbt.ui.printer.print_timestamped_line('Done.')
+
+
+class RemoteCompileTask(CompileTask, RemoteCallable):
+    METHOD_NAME = 'compile'
+
+    def __init__(self, args, config):
+        super(CompileTask, self).__init__(args, config)
+        self.parser = None
+        self._base_manifest = GraphLoader.load_all(
+            config,
+            internal_manifest=get_adapter(config).check_internal_manifest()
+        )
+
+    def get_runner_type(self):
+        return RPCCompileRunner
+
+    def runtime_cleanup(self, selected_uids):
+        """Do some pre-run cleanup that is usually performed in Task __init__.
+        """
+        self.run_count = 0
+        self.num_nodes = len(selected_uids)
+        self.node_results = []
+        self._skipped_children = {}
+        self._skipped_children = {}
+        self._raise_next_tick = None
+
+    def handle_request(self, name, sql, timeout=None):
+        self.parser = RPCCallParser(
+            self.config,
+            all_projects=load_all_projects(self.config),
+            macro_manifest=self._base_manifest
+        )
+
+        sql = self.decode_sql(sql)
+        request_path = os.path.join(self.config.target_path, 'rpc', name)
+        node_dict = {
+            'name': name,
+            'root_path': request_path,
+            'resource_type': NodeType.RPCCall,
+            'path': name+'.sql',
+            'original_file_path': 'from remote system',
+            'package_name': self.config.project_name,
+            'raw_sql': sql,
+        }
+        unique_id, node = self.parser.parse_sql_node(node_dict)
+
+        self.manifest = ParserUtils.add_new_refs(
+            manifest=self._base_manifest,
+            current_project=self.config,
+            node=node
+        )
+        # don't write our new, weird manifest!
+        self.linker = compile_manifest(self.config, self.manifest, write=False)
+        selected_uids = [node.unique_id]
+        self.runtime_cleanup(selected_uids)
+        self.job_queue = self.linker.as_graph_queue(self.manifest,
+                                                    selected_uids)
+
+        # TODO: how can we get a timeout in here? Spin off a subprocess?
```

Comment:
This is a good question, and one which I think @cmcarthur would be well suited to help think through

Reply:
oops, need to remove this TODO.